### PR TITLE
Fixes super. function and var resolving

### LINKED ISSUE #114: super. function calls not resolved correctly
For example, super.fromParent2() is not resolved. 

        contract Parent1 {
            function fromParent1() {
            }
        }

        contract Parent2 {
            function fromParent2() {
            }
        }

        contract Child is Parent1, Parent2 {
            function doit() {
                super.fromParent2();
            }
        }

Function calls are resolved correctly only for the first parent (Parent1 in this example)
Will try to fix this

### PR DESCRIPTION
**Pull request by 0v1se**
fix #114

### CODE REVIEW

File: src/main/kotlin/me/serce/solidity/lang/types/types.kt [hunk 1]
```diff
@@ -60,6 +60,10 @@ data class SolContract(val ref: SolContractDefinition) : SolType {
   override fun toString() = ref.name ?: ref.text ?: "$ref"
 }
 
+data class SolSuper(val ref: SolContractDefinition): SolType {
+  override fun toString() = "${ref.name ?: ref.text ?: ref}.super"
+}
```
[the remark this answers is not in the document]
Comment (0v1se) [reply src/main/kotlin/me/serce/solidity/lang/types/types.kt]: Yes, I get what you're talking about. Will propose new way of doing this

**Comment by 0v1se** [comment]
Hey @SerCeMan 
I removed SolSuper, cause it's not type really (was not sure about this at the beginning too)

implemented custom resolving for functions and variables right in methods `resolveMemberAccess` and `SolFunctionCallReference.multiResolve`

other changes: 
- `inferRefType` now returns SolUnknown for "super". I think it's correct behaviour
- `resolveVarLiteral` now returns all possible super contracts (not the first one as was before)

this way, use can go to any super contract if he wants:
image

I didn't find a good solution for unifying "super" for now. So, we have different handling of supers depending on what we need.

Some more concerns:

- I think it's good to remove `SolResolver.resolveTypeName` and replace it with `SolResolver.resolveTypeNameUsingImports` cause it's more precise
- Maybe we should add special literals for `this` and `super`? (to make them bold and to replace "super" and "this" constants from plugin's source code)

**Comment by 0v1se** [comment]
Hey @SerCeMan!
What do you think about the PR? Did you have time to review it?

**Comment by SerCeMan** [comment]
> Did you have time to review it?

Sorry, the first message slipped through the inbox. 

> I think it's good to remove SolResolver.resolveTypeName and replace it with SolResolver.resolveTypeNameUsingImports cause it's more precise

Sounds good to me!

> Maybe we should add special literals for this and super? (to make them bold and to replace "super" and "this" constants from plugin's source code)

It might not be easy because the following is a valid construct
```solidity
contract Ballot {
    function a() public pure returns (uint8) {
        uint8 this = 42;
        return this;
    }
}
```

**Pull request merged**